agent/acl_test.go

```diff
@@ -40,9 +52,9 @@ func (m *MockServer) GetPolicy(args *structs.ACLPolicyRequest, reply *structs.AC
 
 func TestACL_Version8(t *testing.T) {
 	t.Parallel()
-	cfg := TestConfig()
-	cfg.ACLEnforceVersion8 = Bool(false)
-	a := NewTestAgent(t.Name(), cfg)
+	a := NewTestAgent(t.Name(), `
+ 		acl_enforce_version_8 = true
```

Comment:
The default is set to a wrong key `acl_enforce_version8` (not the lacking final underscore) which may contribute to the test failing. 

I am working on a patch which detects invalid field names again and restores the previous functionality. However, since `encoding/json` and `hashicorp/hcl` decode embedded structs differently (`map[string]interface{}` vs `[]map[string]interface{}`). Therefore, just using `mapstructure` as before to detect unused fields won't work. This is a bug in HCL itself for which there is no easy fix AFAICT. Also, using the embedded JSON parser in `hcl` will also return `[]map[string]interface{}` which will trip up `mapstructure`. https://github.com/hashicorp/hcl/issues/209

Unless there is an option in mapstructure to do the mapping I think there are several viable approaches:

1. Patching all `[]map` to `map` and then use `mapstructure` for the mapping and invalid field detection. However, `watches` has `[]map[string]interface{}` so there would need to be an exception for that field and others in the future.

1. Build a custom invalid field detector which given a `map[string]interface{}` and struct type+tag name can detect which fields will not be mappable. 

1. Enhance the `hashicorp/hcl` decoder with an invalid field detector and do the same thing with `encoding/json` by first forking it and then adding that to the custom code until https://github.com/golang/go/issues/15314 is added to the stdlib.

1. Add that capability to `mapstructure`.

For the 1.0 milestone, the first option seems to be the one with the least amount of work so I'll try that first.